modules/xmpp/TraceablePeerConnection.js

```diff
@@ -583,18 +616,35 @@ TraceablePeerConnection.prototype.getStats = function(callback, errback) {
  * - groups - Array of the groups associated with the stream.
  */
 TraceablePeerConnection.prototype.generateNewStreamSSRCInfo = function () {
+    let ssrcInfo = {ssrcs: [], groups: []};
     if (!this.session.room.options.disableSimulcast
         && this.simulcast.isSupported()) {
-        var ssrcInfo = {ssrcs: [], groups: []};
-        for(var i = 0; i < SIMULCAST_LAYERS; i++)
-            ssrcInfo.ssrcs.push(RandomUtil.randomInt(1, 0xffffffff));
+        for (let i = 0; i < SIMULCAST_LAYERS; i++) {
+            ssrcInfo.ssrcs.push(SDPUtil.generateSsrc());
+        }
         ssrcInfo.groups.push({
             primarySSRC: ssrcInfo.ssrcs[0],
             group: {ssrcs: ssrcInfo.ssrcs.join(" "), semantics: "SIM"}});
-        return ssrcInfo;
+        ssrcInfo;
     } else {
-        return {ssrcs: [RandomUtil.randomInt(1, 0xffffffff)], groups: []};
+        ssrcInfo = {ssrcs: [SDPUtil.generateSsrc()], groups: []};
+    }
+    if (!this.session.room.options.disableRtx) {
+        let currNumSsrcs = ssrcInfo.ssrcs.length;
+        for (let i = 0; i < currNumSsrcs; ++i) {
```

Comment:
actually i had to change it back...we add to the list being iterated over here, so i explicitly grabbed the original length so we only iterated over the original items.  i added a comment here too to make that more clear.